tests/kafkatest/services/streams.py

```diff
@@ -466,6 +466,15 @@ def __init__(self, test_context, kafka, processingMode):
                                                                 "org.apache.kafka.streams.tests.BrokerCompatibilityTest",
                                                                 processingMode)
 
+    def prop_file(self):
+        properties = {streams_property.STATE_DIR: self.PERSISTENT_ROOT,
+                      streams_property.KAFKA_SERVERS: self.kafka.bootstrap_servers(),
+                      # the old broker (< 2.4) does not support configuration replication.factor=-1
+                      "replication.factor": 1}
```

Comment:
Should we set it to 3 instead? IIRC, we run all system tests with 3 brokers? Wondering why the change broke the system tests, as they should have overwritten the default to 3 anyway?

Reply:
Ah, yeah, I bet that's it: we only set it for the EOS tests. Might be better to just set it in the Java code instead as it's easier to find and read, and I believe most other configs are set there. I think this test runs the `StreamsSmokeTest`?

Reply:
> Should we set it to 3 instead? IIRC, we run all system tests with 3 brokers?

not really. `streams_broker_compatibility_test.py` run test with single broker ( https://github.com/apache/kafka/blob/trunk/tests/kafkatest/tests/streams/streams_broker_compatibility_test.py#L45)

> Might be better to just set it in the Java code instead as it's easier to find and read, and I believe most other configs are set there. 

I prefer to change python code rather than java code since the number of brokers is connected to replication refactor. If we add hardcode (i.e `replication refactor = 1`) in the java class, it is hard to change both of them in python.

>  I think this test runs the StreamsSmokeTest?

BrokerCompatibilityTest (https://github.com/apache/kafka/blob/trunk/tests/kafkatest/services/streams.py#L466)

Reply:
Fine with me either way. Thanks for the fix

Reply:
Fine with me, too. Feel free to merge.